File: app/src/main/java/com/example/bmicalculator/BmiFragment.java
package com.example.bmicalculator;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class BmiFragment extends Fragment {

    private EditText waga, wzrost;
    private TextView wynik;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        View view = inflater.inflate(R.layout.fragment_bmi, container, false);

        waga = view.findViewById(R.id.waga);
        wzrost = view.findViewById(R.id.wzrost);
        wynik = view.findViewById(R.id.wynik);
        Button oblicz = view.findViewById(R.id.oblicz);

        oblicz.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                obliczBMI();
            }
        });
        return view;
    }

    private void obliczBMI() {

        String S1 = waga.getText().toString();
        String S2 = wzrost.getText().toString();

        float wagaValue = Float.parseFloat(S1);
        float wzrostValue = Float.parseFloat(S2) / 100;

        float bmi = wagaValue / (wzrostValue * wzrostValue);
        bmi = Math.round(bmi);

        printBMI(bmi);
    }

    private void printBMI(float bmi) {

        String wynikBMI;

        if(bmi < 16) {
            wynikBMI = "Wygłodzenie";
        } else if(bmi >= 16 && bmi < 17) {
            wynikBMI = "Wychudzenie";
        } else if(bmi >= 17 && bmi < 18.5) {
            wynikBMI = "Niedowaga";
        } else if(bmi >= 18.5 && bmi < 25) {
            wynikBMI = "Waga prawidłowa";
        } else if(bmi >= 25 && bmi < 30) {
            wynikBMI = "Nadwaga";
        } else if(bmi >= 30 && bmi < 35) {
            wynikBMI = "Otyłość I stopnia";
        } else if(bmi >= 35 && bmi < 40) {
            wynikBMI = "Otyłość II stopnia";
        } else {
            wynikBMI = "Otyłość III stopnia";
        }

        String obliczenia = bmi + "\n" + wynikBMI;

        wynik.setText(obliczenia);
    }
}

File: app/src/main/java/com/example/bmicalculator/kotlinclasses/BmiFragment.kt
package com.example.bmicalculator.kotlinclasses;

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import com.example.bmicalculator.R
import kotlin.math.roundToInt
import kotlinx.android.synthetic.main.fragment_bmi.*

class BmiFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val view = inflater.inflate(R.layout.fragment_bmi, container, false)
        val oblicz = view.findViewById<Button>(R.id.oblicz)
        oblicz.setOnClickListener { obliczBMI() }

        return view
    }

    private fun obliczBMI() {

        if(waga.text.isNullOrEmpty() || wzrost.text.isNullOrEmpty()) {
            wynik!!.text = getString(R.string.podaj_waga_wzrost)
        } else {
            val S1 = waga!!.text.toString()
            val S2 = wzrost!!.text.toString()
            val wagaValue = S1.toFloat()
            val wzrostValue = S2.toFloat() / 100
            var bmi = wagaValue / (wzrostValue * wzrostValue)
            bmi = bmi.roundToInt().toFloat()
            printBMI(bmi)
        }
    }

    private fun printBMI(bmi: Float) {

        val wynikBMI: String

        wynikBMI = if (bmi < 16) {
            "Wygłodzenie"
        } else if (bmi >= 16 && bmi < 17) {
            "Wychudzenie"
        } else if (bmi >= 17 && bmi < 18.5) {
            "Niedowaga"
        } else if (bmi >= 18.5 && bmi < 25) {
            "Waga prawidłowa"
        } else if (bmi >= 25 && bmi < 30) {
            "Nadwaga"
        } else if (bmi >= 30 && bmi < 35) {
            "Otyłość I stopnia"
        } else if (bmi >= 35 && bmi < 40) {
            "Otyłość II stopnia"
        } else {
            "Otyłość III stopnia"
        }

        val obliczenia = bmi.toString() + "\n" + wynikBMI
        wynik!!.text = obliczenia
    }
}
